Analysis/Tasks/PWGCF/FemtoDream/include/FemtoDream/FemtoDreamV0Selection.h

```diff
@@ -0,0 +1,351 @@
+  const static float CPAV0Min = getMinimalSelection(femtoDreamV0Selection::kCPAV0Min, femtoDreamSelection::kLowerLimit);
+  const static float TranRadV0Min = getMinimalSelection(femtoDreamV0Selection::kTranRadV0Min, femtoDreamSelection::kLowerLimit);
+  const static float TranRadV0Max = getMinimalSelection(femtoDreamV0Selection::kTranRadV0Max, femtoDreamSelection::kUpperLimit);
+  const static float DecVtxMax = getMinimalSelection(femtoDreamV0Selection::kDecVtxMax, femtoDreamSelection::kAbsUpperLimit);
+  const static float InvMassV0Max = getMinimalSelection(femtoDreamV0Selection::kInvMassV0Max, femtoDreamSelection::kUpperLimit);
+  const static float InvMassRejMax = getMinimalSelection(femtoDreamV0Selection::kInvMassRejMax, femtoDreamSelection::kUpperLimit);
+
+  if (nPtV0MinSel > 0 && pT < pTV0Min) {
+    return false;
+  }
+  if (nPtV0MaxSel > 0 && pT > pTV0Max) {
+    return false;
+  }
+  if (nDCAV0DaughMax > 0 && dcaDaughv0 > DCAV0DaughMax) {
+    return false;
+  }
+  if (nCPAV0Min > 0 && cpav0 < CPAV0Min) {
+    return false;
+  }
+  if (nTranRadV0Min > 0 && tranRad < TranRadV0Min) {
+    return false;
+  }
+  if (nTranRadV0Max > 0 && tranRad > TranRadV0Max) {
+    return false;
+  }
+  for (int i = 0; i < decVtx.size(); i++) {
+    if (nDecVtxMax > 0 && decVtx.at(i) > DecVtxMax) {
+      return false;
+    }
+  }
+  if (nInvMassV0Max > 0 && std::abs(mMassPDGV0 - mMassV0) > InvMassV0Max) {
+    return false;
+  }
+  if (nInvMassRejMax > 0 && std::abs(mMassPDGRej - mMassRej) > InvMassRejMax) {
+    return false;
+  }
+  return true;
+}
+
+template <typename T1, typename T2>
+void FemtoDreamV0Selection::setChildCuts(femtoDreamV0Selection::ChildTracks child, T1 selVal, T2 selVar, femtoDreamSelection::SelectionType selType)
+{
+  if (child == femtoDreamV0Selection::kPosDaughTrack)
+    PosDaughTrack.setSelection(selVal, selVar, selType);
+  else if (child == femtoDreamV0Selection::kNegDaughTrack)
+    NegDaughTrack.setSelection(selVal, selVar, selType);
+}
+
+/// the CosPA of V0 needs as argument the posXYZ of collisions vertex so we need to pass the collsion as well
+template <typename C, typename V, typename T>
+std::vector<uint64_t> FemtoDreamV0Selection::getCutContainer(C const& col, V const& v0, T const& posTrack, T const& negTrack)
+{
+  posTrack = PosDaughTrack;
+  negTrack = NegDaughTrack;
+  uint64_t outputPosTrack = posTrack.getCutContainer(posTrack);
+  uint64_t outputNegTrack = negTrack.getCutContainer(negTrack);
+  uint64_t output = 0;
+  size_t counter = 0;
+  const auto signPos = posTrack.sign();
+  const auto signNeg = negTrack.sign();
+  if (signPos < 0 || signNeg > 0) {
+    printf("ERROR - Wrong sign for V0 daughters");
+  }
+  const auto pT = v0.pt();
+  const auto tranRad = v0.v0radius();
+  const auto dcaDaughv0 = v0.dcaV0daughters();
+  const auto cpav0 = v0.v0cosPA(col.posX(), col.posY(), col.posZ());
+  const std::vector<float> decVtx = {v0.x(), v0.y(), v0.z()};
+
+  switch (mSelV0) {
```

Comment:
also here the task should be capable of doing multiple particles at the same time